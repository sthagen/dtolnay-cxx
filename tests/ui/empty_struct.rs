#![allow(unexpected_cfgs)]

#[cxx::bridge]
mod ffi {
    struct Empty {}
}

#[cxx::bridge]
mod ffi2 {
    struct ConditionallyEmpty {
        #[cfg(target_os = "nonexistent")]
        never: u8,
        #[cfg(target_os = "another")]
        another: u8,
    }
}

fn main() {}
