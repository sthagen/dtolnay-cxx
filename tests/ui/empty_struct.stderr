error: structs without any fields are not supported
 --> tests/ui/empty_struct.rs:5:5
  |
5 |     struct Empty {}
  |     ^^^^^^^^^^^^^^^

error: structs without any fields are not supported
  --> tests/ui/empty_struct.rs:10:5
   |
10 |     struct ConditionallyEmpty {
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^
